src/Chainweb/Pact/Backend/Bench.hs

```diff
@@ -56,9 +65,78 @@ cpBench = C.envWithCleanup setup teardown $ \ ~(NoopNFData (e,_)) -> C.bgroup "c
       [
         benchUserTable dbEnv "usertable"
       ]
+-}
+
+cpBenchOverBlock :: Int -> C.Benchmark
+cpBenchOverBlock transactionCount = C.envWithCleanup setup teardown $ \ ~(NoopNFData (_,e,_)) -> C.bgroup ("batchedCheckpointer/transactionCount=" ++ show transactionCount) (benches e)
+  where
+
+    setup = do
+      (f,deleter) <- newTempFile
+      !sqliteEnv <- openSQLiteConnection f chainwebPragmas
+      let nolog = newLogger neverLog ""
+      !cenv <- initRelationalCheckpointer initBlockState sqliteEnv nolog freeGasEnv
+      return $ NoopNFData (sqliteEnv, cenv, deleter)
 
-pactSqliteBench :: C.Benchmark
-pactSqliteBench = C.envWithCleanup setup teardown $ \ ~(NoopNFData (e,_)) -> C.bgroup "pact-sqlite" (benches e)
+    teardown (NoopNFData (sqliteEnv, _cenv, deleter)) = do
+        closeSQLiteConnection sqliteEnv
+        deleter
+
+    benches :: CheckpointEnv -> [C.Benchmark]
+    benches cpenv =
+      [
+        datbench cpenv "usertable"
+      ]
+      where
+
+        datbench cp name = C.env (setup' cp) $ \ ~(ut) -> C.bench name $ C.nfIO (go cp ut)
+
+        setup' CheckpointEnv{..} = do
+          usertablename <- restore _cpeCheckpointer Nothing >>= \case
+              PactDbEnv' db@(PactDbEnv pdb e) -> do
+                let tn = "user1"
+                    ut = UserTables tn
+                begin db
+                _createUserTable pdb tn "someModule" e
+                writeRow db Insert ut 1
+                commit db
+                return $ NoopNFData ut
+          save _cpeCheckpointer hash01
+          return usertablename
+
+
+        f = "f"
+        k = "k"
+
+        hash01 = BlockHash $ unsafeMerkleLogHash "0000000000000000000000000000001a"
+        hash02 = BlockHash $ unsafeMerkleLogHash "0000000000000000000000000000002a"
+
+        writeRow (PactDbEnv pdb e) writeType ut i =
+            _writeRow pdb writeType ut k (ObjectMap $ M.fromList [(f,(PLiteral (LInteger i)))]) e
+
+        go CheckpointEnv{..} (NoopNFData ut) = do
+            result <- restore _cpeCheckpointer (Just (BlockHeight 1, hash01)) >>= \case
```

Comment:
Interesting, so this benchmark does a rewind each time. We should make a variant that has an MVar with an increasing height/hash to test "non-rewind performance" too.

Reply:
Done